base/float.jl

```diff
@@ -494,17 +496,29 @@ abs(x::Float64) = box(Float64,abs_float(unbox(Float64,x)))
 
 Test whether a floating point number is not a number (NaN).
 """
+isnan{T<:IEEEFloat}(x::T) = (reinterpret(Unsigned, x) & ~sign_mask(T)) > exponent_mask(T)
```

Comment:
See https://gist.github.com/musm/8150451b04120ef27974d8270dedbf6f 
for more comprehensive tests   (the gist is we are comparing performance in the 10-100 picosecond range...) So basically these instructions are so cheap and fast that it really doesn't matter.

For `isnan` I agree, unfortunately we still require special handling for `Float16` which  prompted this pr, which I thought would be good  to unify all the exceptional case handling  to independent of float hardware instructions so that Float16 would be handled. But the problem there is that cpu support for small floats is not great.

I'll leave this to @vchuravy  expertise on the issue on whether these proposed changes are positive or not, for now I am closing this.